widget/entry.go

```diff
@@ -1009,16 +1009,17 @@ func (e *Entry) getRowCol(p fyne.Position) (int, int) {
 	e.propertyLock.RLock()
 	defer e.propertyLock.RUnlock()
 
-	rowHeight := e.textProvider().charMinSize(e.Password, e.TextStyle).Height
+	text := e.textProvider()
+	rowHeight := text.charMinSize(e.Password, e.TextStyle).Height
 	row := int(math.Floor(float64(p.Y+e.scroll.Offset.Y-theme.LineSpacing()) / float64(rowHeight)))
 	col := 0
 	if row < 0 {
 		row = 0
-	} else if row >= e.textProvider().rows() {
-		row = e.textProvider().rows() - 1
-		col = e.textProvider().rowLength(row)
+	} else if row >= text.rows() {
```

Comment:
Should we call text.rows() just once?

Reply:
I checked and doesn't matter. It us just returning the length of a slice and that method will be inlined.